rdkit/ML/Scoring/Scoring.py

```diff
@@ -14,6 +14,40 @@
 """
 
 import math
+import random  # Does rdkit depend on numpy? => Use np.random instead
+import copy
+from operator import itemgetter
+
+
+def avoid_sorting_artifacts_on_ranking(scores, col, reverse=False):
```

Comment:
I agree. The function name, and some other things I include in the PR (like overly verbose expectations on the unit tests, and even testing there for wrong results) are like they are to help reviewers understand the rationale of the PR. They won't survive when the PR is ready to merge.

If we make "sortByColumns" a function that can be used somewhere else, then I agree with allowing to pass more than one column to ignore. Otherwise we should maybe, for consistency with the rest of the Metrics module, stick with allowing only one column to ignore. Note that I named `col` to follow the convention of the rest of the module. I agree, it should be renamed to something like `classCol` all around the module - but that in python is an API breaking change.

Oh well, naming, one of the two harder things in programming, together with cache invalidation and off-by-one errors.